src/io/memory.cr

```diff
@@ -370,3 +370,10 @@ class MemoryIO
     @buffer = @buffer.realloc(@capacity)
   end
 end
+
+class MemoryIO < IO::Memory
+  def self.new(*args, **nargs)
+    {{ puts "Warning: MemoryIO is deprecated and will be removed after 0.20.0, use IO::Memory instead".id }}
```

Comment:
There's no way to get that info right now. In the future we could add it, but it will always be the one of the first call site, because later method calls with the same argument types are not re-analyzed